handsontable/test/e2e/Core_setDataAtCell.spec.js

```diff
@@ -317,6 +317,71 @@ describe('Core_setDataAtCell', () => {
     expect(errors).toBe(0);
   });
 
+  it('should trigger `beforeChange` and `afterSetDataAtCell` in the correct order', () => {
+    let hooksOrder = '';
+
+    handsontable({
+      data: [[1, 1]],
+      beforeChange() {
+        hooksOrder += '1';
+      },
+      afterSetDataAtCell() {
+        hooksOrder += '2';
+      },
+    });
+
+    setDataAtCell(0, 0, 5);
+
+    expect(hooksOrder).toBe('12');
```

Comment:
```suggestion
    const beforeChange = jasmine.createSpy('beforeChange');
    const afterSetDataAtCell = jasmine.createSpy('afterSetDataAtCell');

    handsontable({
      data: [[1, 1]],
      beforeChange,
      afterSetDataAtCell,
    });

    setDataAtCell(0, 0, 5);

    expect(beforeChange).toHaveBeenCalledBefore(afterSetDataAtCell);
```
May I suggest using the `toHaveBeenCalledBefore` matcher here? In case when the test fails, the description will be more descriptive than the current one (value `12` does not match the `21`).